source/PlayerInfo.cpp

```diff
@@ -2016,6 +2018,54 @@ void PlayerInfo::SetTravelDestination(const Planet *planet)
 
 
 
+void PlayerInfo::QueueTeleport(const Planet *teleportPlanet)
+{
+	teleportDestination = teleportPlanet;
+}
+
+
+
+void PlayerInfo::DoQueuedTeleport()
+{
+	if(teleportDestination && planet)
+	{
+		system = teleportDestination->GetSystem();
+		planet = teleportDestination;
+		flagship->SetSystem(system);
+		flagship->SetPlanet(planet);
+		for(const shared_ptr<Ship> &ship : ships)
+		{
+			if(!ship->IsParked())
```

Comment:
We probably also need to ignore destroyed  ships? And maybe with an option we should only tp those in this current ssysem the player is in

Reply:
I think we should always teleport ship that are in the same system.
done